### ISSUE TITLE
Enum binding when it shouldn't

### ISSUE THREAD

**Issue by Mardoxx** [aspnet/Mvc#7031]
If we have the following:

```
public enum TestEnum {
    Val1 = 1,
    Val2 = 2
}
```
```
[HttpGet("test/{testEnum}")]
public IActionResult Get(
    [FromRoute(Name = "testEnum")] TestEnum test,
{
    return Ok(test);
}
```

`GET test/10` will return `10`, and likewise `GET test/invalid` returns `0` when I should not expect either to bind.

How can I configure it to do so? I assume I need to set `allowBindingUndefinedValueToEnumType` some how? https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/EnumTypeModelBinder.cs

**Comment by Mardoxx** [comment]
Ah.. I see this was added here https://github.com/aspnet/Mvc/pull/6812

And there's this potentially related issue https://github.com/aspnet/Mvc/issues/6858

Will this be the correct way to configure this?

**Comment by Eilon** [comment]
@kichalla can you take a look?

**Comment by kichalla** [comment]
This issue is a duplicate of the issue: https://github.com/aspnet/Mvc/issues/6329. (Issue https://github.com/aspnet/Mvc/issues/6858 is a side effect of the fix that was made for it)

@Mardoxx The fix you suggested is part of 2.1.0-preview1 and is only available on MyGet though.

**Comment by Mardoxx** [comment]
Great thanks!
